sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -31,10 +31,11 @@ import org.apache.spark.sql.catalyst.util.{CaseInsensitiveMap, CompressionCodecs
  * Most of these map directly to Jackson's internal options, specified in [[JsonParser.Feature]].
  */
 private[sql] class JSONOptions(
-    @transient private val parameters: CaseInsensitiveMap)
+    @transient private val parameters: CaseInsensitiveMap, defaultTimeZoneId: String)
```

Comment:
Ah, yes, it needed to introduce such logics below before creating `JSONOptions`/`CSVOptions`.
```scala
val options = extraOptions.toMap
val caseInsensitiveOptions = new CaseInsensitiveMap(options)
if (caseInsensitiveOptions.contains("timeZone")) {
  caseInsensitiveOptions
} else {
  new CaseInsensitiveMap(
  options + ("timeZone" -> sparkSession.sessionState.conf.sessionLocalTimeZone))
}

val parsedOptions: JSONOptions = new JSONOptions(optionsWithTimeZone)
```

So, I suggested this way as It seems also because the default value of `timeZone` can be varied. It seems `ParquetOptions` also takes another argument for the same reason.

Another way I suggested is, to make this `Option[TimeZone]` to decouple the variant of the default value (like `JSONOptions.columnNameOfCorruptRecord`) but it seems `timestampFormat` in both options are dependent on `timeZone`. In that case, we should make it `Option` too which seems introducing some more complexity. So, it seems above way is better.

I am fine if we find a better cleaner way.